File: examples-sdk/swift/006-basic-shm-extractor/main.swift
// 006-basic-shm-extractor — Swift SDK equivalent
// Compile: vil compile --from swift --input 006-basic-shm-extractor/main.swift --release

let server = VilServer(name: "shm-extractor-demo", port: 8080)
let shm_demo = ServiceProcess(name: "shm-demo")
shm_demo.endpoint(method: "POST", path: "/ingest", handler: "ingest")
shm_demo.endpoint(method: "POST", path: "/compute", handler: "compute")
shm_demo.endpoint(method: "GET", path: "/shm-stats", handler: "shm_stats")
shm_demo.endpoint(method: "GET", path: "/benchmark", handler: "benchmark")
server.service(shm_demo)
server.compile()

File: examples-sdk/csharp/006-basic-shm-extractor/Main.cs
// 006-basic-shm-extractor — C# SDK equivalent
// Compile: vil compile --from csharp --input 006-basic-shm-extractor/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("shm-extractor-demo", 8080);
var shm_demo = new ServiceProcess("shm-demo");
shm_demo.Endpoint("POST", "/ingest", "ingest");
shm_demo.Endpoint("POST", "/compute", "compute");
shm_demo.Endpoint("GET", "/shm-stats", "shm_stats");
shm_demo.Endpoint("GET", "/benchmark", "benchmark");
server.Service(shm_demo);
server.Compile();
